src/Octoshift/Services/ReclaimService.cs

```diff
@@ -0,0 +1,189 @@
+﻿using System;
+using System.Linq;
+using System.Threading.Tasks;
+using Octoshift.Models;
+using OctoshiftCLI;
+using OctoshiftCLI.Models;
+
+namespace Octoshift.Services
+{
+    public class ReclaimService
+    {
+        private readonly GithubApi _githubApi;
+        private readonly OctoLogger _log;
+        private bool _failed;
+
+        public ReclaimService(GithubApi githubApi, OctoLogger logger)
+        {
+            _githubApi = githubApi;
+            _log = logger;
+        }
+        public virtual async Task ReclaimMannequin(string mannequinUser, string mannequinId, string targetUser, string githubOrg, bool force)
+        {
+            var githubOrgId = await _githubApi.GetOrganizationId(githubOrg);
+
+            var mannequins = new Mannequins((await GetMannequins(githubOrgId)).GetByLogin(mannequinUser, mannequinId));
+            if (mannequins.Empty())
+            {
+                throw new OctoshiftCliException($"User {mannequinUser} is not a mannequin.");
+            }
+
+            // (Potentially) Save one call to the API to get the targer user
+            // // (it also makes it unnecessary to check for claimed users during reclaiming loop)
+            if (!force && mannequins.IsClaimed(mannequinUser))
+            {
+                throw new OctoshiftCliException($"User {mannequinUser} is already mapped to a user. Use the force option if you want to reclaim the mannequin again.");
+            }
+
+            var targetUserId = await _githubApi.GetUserId(targetUser);
+            if (targetUserId == null)
+            {
+                throw new OctoshiftCliException($"Target user {targetUser} not found.");
+            }
+
+            _failed = false;
+
+            // get all unique mannequins by login and id and map them all to the same target
+            foreach (var mannequin in mannequins.UniqueUsers())
+            {
+                var result = await _githubApi.ReclaimMannequin(githubOrgId, mannequin.Id, targetUserId);
+
+                HandleResult(mannequinUser, targetUser, mannequin, targetUserId, result);
+            }
+
+            // Fail if there was at least one error
+            HandleResult();
```

Comment:
It's more clear to simply check for failure and throw an exception right here instead of using a method.